### ISSUE TITLE
Register - unhandled error

### ISSUE THREAD

**Issue by delvedor** [fastify/fastify#59]
While working at https://github.com/fastify/point-of-view/pull/1 I found that if I pass and error to the `next` callback of `register`, even if I handle it, fastify will throw and error.
This because of [this](https://github.com/mcollina/avvio/blob/master/boot.js#L96) specific line of avvio.
Internally we have not a listener for that error event, and node will throw the following error:
```
< Error: Missing engine
<     at fastifyView (.../point-of-view/index.js:13:10)
<     at Plugin.exec (.../point-of-view/node_modules/avvio/boot.js:191:3)
<     at Boot.loadPlugin (.../point-of-view/node_modules/avvio/boot.js:205:10)
<     at release (.../point-of-view/node_modules/fastq/queue.js:127:16)
<     at Object.resume (.../point-of-view/node_modules/fastq/queue.js:61:7)
<     at _combinedTickCallback (internal/process/next_tick.js:67:7)
<     at process._tickCallback (internal/process/next_tick.js:98:9)
<     at Timeout.Module.runMain [as _onTimeout] (module.js:606:11)
<     at ontimeout (timers.js:365:14)
<     at tryOnTimeout (timers.js:237:5)
< events.js:160
<       throw er; // Unhandled 'error' event
<     
```
This can be easily fixed with the following line inside Fastify.
```js
app.on('error', () => {})
```

Now the question is, how do we want to solve this issue?
The faster solution is to update avvio, remove the error emitter and update the `noop` function is this way:
```js
function noop (err) {
  if (err) throw err
}
```
But I think that in the scope of avvio as is, that error emitter is useful to catch global errors.
Contrariwise, in Fastify it does not make sense imo, because is counterintuitive for the user.

Thoughts?
cc @mcollina

**Comment by mcollina** [comment]
I think we should give this error to the user, as the first error of the listen callback, or it should throw when that happens.

**Comment by delvedor** [comment]
I like use the listen callback, but my concern is that even if the user has a callback in the register that will throw, this error will arrive in any case to the emitter.

Example:
```js
fastify.register(require('will-throw'), {}, err => {
  if (err) console.log(err)
})

fastify.listen(3000, err => {
  // err will be the same as above
})
```
I don't think this is a good behavior, as said above is pretty counterintuitive.

**Comment by mcollina** [comment]
That's fine. Listen cannot go through if the first one is skipped. All the plugins needs to initialize correctly.

**Comment by delvedor** [comment]
Closed by #61
